napari/_vispy/utils/quaternion.py

```diff
@@ -24,34 +23,19 @@ def quaternion2euler(quaternion, degrees=False):
     angles : 3-tuple
         Euler angles in (rx, ry, rz) order.
     """
-    epsilon = 1e-10
-
     q = quaternion
-
-    sin_theta_2 = 2 * (q.w * q.y - q.z * q.x)
-    sin_theta_2 = np.sign(sin_theta_2) * min(abs(sin_theta_2), 1)
-
-    if abs(sin_theta_2) > 1 - epsilon:
-        theta_1 = -np.sign(sin_theta_2) * 2 * np.arctan2(q.x, q.w)
-        theta_2 = np.arcsin(sin_theta_2)
-        theta_3 = 0
-
-    else:
-        theta_1 = np.arctan2(
-            2 * (q.w * q.z + q.y * q.x),
-            1 - 2 * (q.y * q.y + q.z * q.z),
+    with warnings.catch_warnings():
```

Comment:
Each time the `warnings.catch_warnings()` is used then a list of already seen warnings is reset. 
So using it in the context of often-used functions may result in bigger spamming of user. Why warning happen and how we avoid it?

Reply:
https://github.com/python/cpython/issues/73858

Reply:
I could send a pr to SciPy with a parameter `gimbal_lock_ok:bool` to prevent the warning.